### PR TITLE
make ufoLib ufo3k work with Py23 

### PR DESCRIPTION
**Pull request by moyogo**

### CODE REVIEW

File: Lib/ufoLib/validators.py [hunk 1]
```diff
@@ -583,7 +592,8 @@ def anchorValidator(value):
 	"""
 	dictPrototype = dict(
 		x=((int, float), False), y=((int, float), False),
-		name=(basestring, False), color=(basestring, False), identifier=(basestring, False)
+		name=(basestring, False), color=(basestring, False),
+        identifier=(basestring, False)
```
Comment (adrientetar) [inline Lib/ufoLib/validators.py]: indent

File: Lib/ufoLib/test/test_glifLib.py [hunk 2]
```diff
@@ -73,20 +85,20 @@ def myGlyphNameToFileName(glyphName, glyphSet):
...
-		for glyphName in src.keys():
+		for glyphName in list(src.keys()):
 			g = src[glyphName]
 			g.drawPoints(None)  # load attrs
 			dst.writeGlyph(glyphName, g, g.drawPoints)
 		d = {}
-		for k, v in src.contents.items():
-			print k, v
+		for k, v in list(src.contents.items()):
+			print(k, v)
```
Comment (adrientetar) [inline Lib/ufoLib/test/test_glifLib.py]: Are you sure that this works with Python 2 without from __future__ import

**Comment by adrientetar** [comment]
tbh all the list(dict.items()) that were put by 2to3 are unnecessary (unless one does indexing), they are put to keep equivalent semantics with Python2. 
Also, maybe we could drop the embedded plistlib entirely and rely on the one in the standard library. Does it do anything that the standard one doesn't?

**Comment by adrientetar** [comment]
OK, I'll pull this branch and later developments can be done later.

**Comment by moyogo** [comment]
We should drop embedded plistlib entirely. The only classes used are PlistParser and PlistWriter which are _PlistParser and _PlistWriter in Python 3.4+. These are used to read and write the plist-like `dict` the various `lib`.

plistlib’s readPlist() and writePlist() are also deprecated in later versions of Python, load() and dump() should be used instead.

**Comment by adrientetar** [comment]
Cool, do you want to drop it now in your branch? That would be great.

File: Lib/ufoLib/__init__.py [hunk 3]
```diff
@@ -28,23 +38,21 @@
 	convertFontInfoValueForAttributeFromVersion3ToVersion2
 """
 
-
-import os
-import shutil
-from cStringIO import StringIO
```
Comment (adrientetar) [inline Lib/ufoLib/__init__.py]: Not using cStringIO on Python 2 risks decreasing performance.

File: Lib/ufoLib/glifLib.py [hunk 4]
```diff
@@ -12,13 +13,13 @@
 """
 
 import os
-from cStringIO import StringIO
+from io import BytesIO, StringIO, open
```
Comment (adrientetar) [inline Lib/ufoLib/glifLib.py]: And here.

File: Lib/ufoLib/plistlib.py [hunk 5]
```diff
@@ -59,7 +59,7 @@
 # Note: the Plist and Dict classes have been deprecated.
 
 import binascii
-from cStringIO import StringIO
+from io import StringIO
```
Comment (adrientetar) [inline Lib/ufoLib/plistlib.py]: And here.

File: Lib/ufoLib/__init__.py (hunk 3 above)
Comment (anthrotype) [inline Lib/ufoLib/__init__.py]: but cStringIO is Python2 only whereas io.BytesIO is both python 2 and 3...

Comment (anthrotype) [inline Lib/ufoLib/__init__.py]: BTW, the `_io` module is written in C in both python2.7 and python3.x

**Comment by moyogo** [comment]
@adrientetar there are some 2-3 bytes-vs-str issues with Python’s PlistWriter. We can fix this later if that’s OK.

**Comment by adrientetar** [comment]
Yea, sure.

**Comment by anthrotype** [comment]
this one can be closed as ufo3k branch has already been splitted into standalone ufoLib.

**Comment by benkiel** [comment]
Closing as per above.

**Pull request closed without merging**